lib/wasi/src/utils.rs

```diff
@@ -1,18 +1,37 @@
 use wasmer_runtime_core::module::Module;
 
+#[allow(dead_code)]
 /// Check if a provided module is compiled with WASI support
 pub fn is_wasi_module(module: &Module) -> bool {
-    if module.info().imported_functions.is_empty() {
-        return false;
-    }
-    for (_, import_name) in &module.info().imported_functions {
-        let namespace = module
-            .info()
-            .namespace_table
-            .get(import_name.namespace_index);
-        if namespace != "wasi_unstable" {
-            return false;
+    get_wasi_version(module) == Some(WasiVersion::Snapshot1)
```

Comment:
I would do it just `.is_some()` ? (both snapshot0 and snapshot1 are wasi modules)

Reply:
I was thinking that I'd attempt to preserve more of the semantics of the function call; either way this is a breaking change -- I think you're right that it doesn't make sense to keep the function name as what it is and not have that behavior.  I'll update it